src/main/java/seedu/main/Main.java

```diff
@@ -0,0 +1,227 @@
+
+    // Singleton TransactionList for use across classes
+    private static TransactionList transactions;
+
+    private static boolean isRunning = true;
+
+    public static void main(String[] args) {
+        scanner = new Scanner(System.in);
+        while (isRunning) {
+            run();
+        }
+    }
+
+    /**
+     * Setter for the chatbot's running state.
+     *
+     * @param isRunning A boolean showing if the chatbot should continue running.
+     */
+    public static void setRunning(boolean isRunning) {
+        Main.isRunning = isRunning;
+    }
+
+    /**
+     * Starts the chatbot and enters the command processing loop.
+     */
+    public static void run() {
+        try {
+            start();
+
+            // Assert that categories and transactions are initialized
+            assert categories != null : "Categories should be initialized.";
+            assert transactions != null : "Transactions should be initialized.";
+
+            runCommandLoop();
+        } catch (Exception e) {
+            logger.log(Level.WARNING, e.getMessage());
+        }
+    }
+
+    /**
+     * Starts the chatbot by printing the logo and greeting messages,
+     * and sign up the Command objects.
+     */
+    public static void start() {
+        logger.log(Level.INFO, "Starting uNivUSaver...");
+
+        scanner = new Scanner(System.in);
+        parser = new Parser();
+        categories = new CategoryList();
+        transactions = new TransactionList();
+
+        HelpCommand helpCommand = new HelpCommand();
+        parser.registerCommands(helpCommand);
+
+        AddCategoryCommand addCategoryCommand = new AddCategoryCommand(categories);
+        parser.registerCommands(addCategoryCommand);
+
+        parser.registerCommands(new AddIncomeCommand(transactions));
+
+        parser.registerCommands(new AddExpenseCommand(transactions));
+
+
+        ViewCategoryCommand viewCategoryCommand = new ViewCategoryCommand(categories);
+        parser.registerCommands(viewCategoryCommand);
+
+        DeleteCategoryCommand deleteCategoryCommand = new DeleteCategoryCommand(categories);
+        parser.registerCommands(deleteCategoryCommand);
+
+        DeleteTransactionCommand deleteTransactionCommand = new DeleteTransactionCommand(transactions);
+        parser.registerCommands(deleteTransactionCommand);
+
+        ViewExpenseCommand viewExpenseCommand = new ViewExpenseCommand(transactions);
+        parser.registerCommands(viewExpenseCommand);
+
+        ViewIncomeCommand viewIncomeCommand = new ViewIncomeCommand(transactions);
+        parser.registerCommands(viewIncomeCommand);
+
+        ViewTotalCommand viewTotalCommand = new ViewTotalCommand(transactions);
+        parser.registerCommands(viewTotalCommand);
+
+        HistoryCommand historyCommand = new HistoryCommand(transactions);
+        parser.registerCommands(historyCommand);
+
+        ByeCommand byeCommand = new ByeCommand();
+        parser.registerCommands(byeCommand);
```

Comment:
You may want to consider refactoring this to a new file/function

Reply:
Created a new issue. Thank you!